pyrgg/functions.py

```diff
@@ -120,28 +120,68 @@ def filesize(fileaddr):  # pragma: no cover
     print("Graph File Size : " + convert_bytes(file_size))
 
 
-def logger(vertices_number, edge_number, file_name, elapsed_time):
+def logger(
+        file_name, 
+        vertices_number, 
+        edge_number, 
+        max_edge, 
+        min_edge, 
+        directed, 
+        signed, 
+        multigraph, 
+        self_loop, 
+        weighted, 
+        max_weight, 
+        min_weight, 
+        elapsed_time):
     """
     Save generated graphs log.
 
+    :param file_name: file name
+    :type file_name: str
     :param vertices_number: number of vertices
     :type vertices_number:int
     :param edge_number: number of edges
     :type edge_number: int
-    :param file_name: file name
-    :type file_name: str
+    :param max_edge: maximum number of edges
+    :type max_edge: int
+    :param min_edge: minimum number of edges
+    :type min_edge: int
+    :param directed: directed
+    :type directed: int
+    :param signed: weight sign flag
+    :type signed: int
+    :param multigraph: multigraph flag
+    :type multigraph: int
+    :param self_loop: self loop flag
+    :type self_loop: int
+    :param weighted: weighted flag
+    :type weighted: int
+    :param max_weight: maximum weight
+    :type max_weight: int
+    :param min_weight: minimum weight
+    :type min_weight: int
     :param elapsed_time: elapsed time
     :type elapsed_time : str
     :return:  None
     """
     try:
         with open("logfile.log", "a") as file:
-            file.write(str(datetime.datetime.now()) + "\n")
-            file.write("Filename : " + file_name + "\n")
-            file.write("Vertices : " + str(vertices_number) + "\n")
-            file.write("Edges : " + str(edge_number) + "\n")
-            file.write("Elapsed Time : " + str(elapsed_time) + "\n")
-            file.write("-------------------------------\n")
+            file.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + "\n" +
+                "Filename : " + file_name + "\n" +
+                "Vertices : " + str(vertices_number) + "\n" +
+                "Total Edges : " + str(edge_number) + "\n" +
+                "Max Edge : " + str(max_edge) + "\n" +
+                "Min Edge : " + str(min_edge) + "\n" +
+                "Directed : " + str(bool(directed)) + "\n" +
+                "Signed : " + str(bool(signed)) + "\n" +
+                "Multigraph : " + str(bool(multigraph)) + "\n" +
+                "Self Loop : " + str(bool(self_loop)) + "\n" +
+                "Weighted : " + str(bool(weighted)) + "\n" +
+                "Max Weight : " + str(max_weight) + "\n" +
+                "Min Weight : " + str(min_weight) + "\n" +
+                "Elapsed Time : " + elapsed_time + "\n" +
+                "-------------------------------\n")
```

Comment:
There was a problem that I fixed. Suppose the `file_name` parameter isn't a `string`. So, the line
```python
file.write("Filename : " + file_name + "\n")
```
raises an error. but the first line of log, which is
```python
file.write(datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S') + "\n")
```
is written in the file. So, I decided to use just one `file.write()` function call to fix this error.

@sepandhaghighi @sadrasabouri